src/coreclr/tools/Common/TypeSystem/Common/MethodDesc.cs

```diff
@@ -234,35 +240,95 @@ private bool Equals(MethodSignature otherSignature, bool allowCovariantReturn)
-                return true;
-            }
-
-            if (this._embeddedSignatureData != null && otherSignature._embeddedSignatureData != null)
+            if (!checkOnlyCallConvention)
             {
-                if (this._embeddedSignatureData.Length != otherSignature._embeddedSignatureData.Length)
+                if (this.EmbeddedSignatureMismatchPermitted ||
+                    otherSignature.EmbeddedSignatureMismatchPermitted)
                 {
-                    return false;
+                    return true;
                 }
 
-                for (int i = 0; i < this._embeddedSignatureData.Length; i++)
+                if (this._embeddedSignatureData != null && otherSignature._embeddedSignatureData != null)
                 {
-                    if (this._embeddedSignatureData[i].index != otherSignature._embeddedSignatureData[i].index)
-                        return false;
-                    if (this._embeddedSignatureData[i].kind != otherSignature._embeddedSignatureData[i].kind)
-                        return false;
-                    if (this._embeddedSignatureData[i].type != otherSignature._embeddedSignatureData[i].type)
+                    if (this._embeddedSignatureData.Length != otherSignature._embeddedSignatureData.Length)
+                    {
                         return false;
+                    }
+
+                    for (int i = 0; i < this._embeddedSignatureData.Length; i++)
+                    {
+                        if (this._embeddedSignatureData[i].index != otherSignature._embeddedSignatureData[i].index)
+                            return false;
+                        if (this._embeddedSignatureData[i].kind != otherSignature._embeddedSignatureData[i].kind)
+                            return false;
+                        if (this._embeddedSignatureData[i].type != otherSignature._embeddedSignatureData[i].type)
+                            return false;
+                    }
+
+                    return true;
                 }
+            }
+            else
+            {
+                if (this.TryGetCallConvModOpts(out EmbeddedSignatureData[] thisModoptData)
+                    == otherSignature.TryGetCallConvModOpts(out EmbeddedSignatureData[] otherModoptData))
+                {
+                    if (thisModoptData == null && otherModoptData == null)
+                    {
+                        return true;
+                    }
 
-                return true;
+                    for (int i = 0; i < thisModoptData.Length; i++)
+                    {
+                        if (thisModoptData[i].index != otherModoptData[i].index)
+                            return false;
+                        if (thisModoptData[i].kind != otherModoptData[i].kind)
+                            return false;
+                        if (thisModoptData[i].type != otherModoptData[i].type)
+                            return false;
+                    }
+                }
```

Comment:
The runtime doesn't mind the ordering of the modifiers.

`SuppressGCTransition, MemberFunction` is treated the same as `MemberFunction, SuppressGCTransition`. We should make them compare the same.

I would really just do the dumb thing with `O(n^2)` complexity I suggested in https://github.com/dotnet/runtime/pull/70105#discussion_r887623570 and for each modopt on the left side, check if the modopt exists on the right side. And make sure the count of interesting modopts is the same.

Reply:
Sigh. Too much care about perf sometime bad